### PR TITLE
added interface support for unique including unittest for it

### PR DESCRIPTION
**Pull request by vushu**
I tried to tinker around with `Unique` and it looks like it doesn't support interfaces, so I tried adding it.
I also tried to do the same for `RefCounted`, but had some problems.

### CODE REVIEW

File: source/automem/unique.d [hunk 1]
```diff
@@ -174,7 +174,7 @@ private:
                 auto repr = (cast(void*)_object)[0..__traits(classInstanceSize, Type)];
                 GC.removeRange(&repr[(void*).sizeof]);
             }();
-        } else static if (supportGC && hasIndirections!Type) {
+        } else static if (supportGC && hasIndirections!Type && !is(Type == interface)) {
```
Comment (atilaneves) [inline source/automem/unique.d:177]: Why is this needed?

**Comment by vushu** [comment]
@atilaneves good question, but it will say boom when we try to `GC.removeRange(_object);` for type of interface.
my best guess is that interfaces can't be dynamically allocated meaning you can't `new`an `interface`, therefore we can't `GC.removeRange.`

**Comment by vushu** [comment]
I have added asserts to ensure that the class gets properly de-allocated.

**Review by atilaneves — APPROVED** [review]
